lib/src/source_visitor.dart

```diff
@@ -2295,6 +2300,110 @@ class SourceVisitor extends ThrowingAstVisitor {
     builder.endSpan();
   }
 
+  @override
+  void visitRecordLiteral(RecordLiteral node) {
+    modifier(node.constKeyword);
+    _visitCollectionLiteral(
+        node, node.leftParenthesis, node.fields, node.rightParenthesis);
+  }
+
+  @override
+  void visitRecordTypeAnnotation(RecordTypeAnnotation node) {
+    var namedFields = node.namedFields;
+
```

Comment:
This is largely copied from _visitTrailingCommaParameterList(), but there are enough differences that it's hard to share code between them.